src/Uno.UI/UI/Xaml/Controls/Repeater/ItemsRepeater.cs

```diff
@@ -0,0 +1,821 @@
+			//if (SharedHelpers.IsRS3OrHigher())
+			{
+				TabFocusNavigation = KeyboardNavigationMode.Once;
+				XYFocusKeyboardNavigation = XYFocusKeyboardNavigationMode.Enabled;
+			}
+
+			Loaded += OnLoaded;
+			Unloaded += OnUnloaded;
+
+			// Initialize the cached layout to the default value
+			var layout = Layout as VirtualizingLayout;
+			OnLayoutChanged(null, layout);
+		}
+
+		~ItemsRepeater()
+		{
+			m_itemTemplate = null;
+			m_animator = null;
+			m_layout = null;
+		}
+
+		#region IUIElementOverrides
+
+		protected override AutomationPeer OnCreateAutomationPeer()
+		{
+			return new RepeaterAutomationPeer(this);
+		}
+
+		#endregion
+
+		#region IUIElementOverrides7
+
+		protected override IEnumerable<DependencyObject> GetChildrenInTabFocusOrder()
+		{
+			return CreateChildrenInTabFocusOrderIterable();
+		}
+
+		#endregion
+
+		#region IUIElementOverrides8
+		protected override void OnBringIntoViewRequested(BringIntoViewRequestedEventArgs e)
+		{
+			m_viewportManager.OnBringIntoViewRequested(e);
+		}
+
+		#endregion
+
+		#region IFrameworkElementOverrides
+
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			if (m_isLayoutInProgress)
+			{
+				throw new InvalidOperationException("Reentrancy detected during layout.");
+			}
+
+			if (IsProcessingCollectionChange)
+			{
+				throw new InvalidOperationException("Cannot run layout in the middle of a collection change.");
+			}
+
+			m_viewportManager.OnOwnerMeasuring();
+
+			m_isLayoutInProgress = true;
+			using var layoutInProgress = Uno.Disposables.Disposable.Create(() => m_isLayoutInProgress = false);
+
+			m_viewManager.PrunePinnedElements();
+			Rect extent = default;
+			Size desiredSize = default;
+
+			var layout = Layout;
+			if (layout != null)
+			{
+				var layoutContext = GetLayoutContext();
+
+				// Expensive operation, do it only in debug builds.
+#if DEBUG && false // TODO UNO
+				var virtualContext = (VirtualizingLayoutContext) layoutContext;
+				virtualContext.Indent(Indent());
+#endif
+
+				// Checking if we have an data template and it is empty
+				if (m_isItemTemplateEmpty)
+				{
+					// Has no content, so we will draw nothing and request zero space
+					extent = new Rect(m_layoutOrigin.X, m_layoutOrigin.Y, 0, 0);
+				}
+				else
+				{
+					desiredSize = layout.Measure(layoutContext, availableSize);
+					extent = new Rect(m_layoutOrigin.X, m_layoutOrigin.Y, desiredSize.Width, desiredSize.Height);
+				}
+
+				// Clear var recycle candidate elements that have not been kept alive by layout - i.e layout did not
+				// call GetElementAt(index).
+				var children = Children;
+				for (var i = 0; i < children.Count; ++i)
+				{
+					var element = children[i];
+					var virtInfo = GetVirtualizationInfo(element);
+
+					if (virtInfo.Owner == ElementOwner.Layout &&
+						virtInfo.AutoRecycleCandidate &&
+						!virtInfo.KeepAlive)
+					{
+						// TODO UNO REPEATER_TRACE_INFO("AutoClear - %d \n", virtInfo.Index);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)